include/crab/domains/symbolic_variable_eq_domain.hpp

```diff
@@ -0,0 +1,1031 @@
+
+    if (!contains(x)) {
+      make_set(x, absval_ptr);
+    } else {
+      // Modify the abstract state of the whole equivalence class
+      element_t rep_x = find(x);
+      equivalence_class_t &ec_x = m_classes.at(rep_x);
+      ec_x.set_absval(absval_ptr);
+    }
+  }
+
+  /// @brief get the domain value stored in current equivalent class
+  /// @param v an element in some set
+  /// @return null if !contains(x), otherwise, returns a shared pointer to value
+  std::shared_ptr<domain_t> get(const element_t &x) {
+    if (is_bottom()) {
+      CRAB_ERROR("called ", domain_name(), "::", __func__, " on bottom");
+    }
+    if (is_top() || !contains(x)) {
+      return nullptr;
+    }
+
+    element_t rep_x = find(x);
+    equivalence_class_t &ec_x = m_classes.at(rep_x);
+    return ec_x.detach_and_get_absval();
+  }
+
+  /// @brief get the equivalent class by giving a element
+  /// @param x an element
+  /// @return if the element exists, return corresponding element class
+  boost::optional<element_set_t> get_variables(const element_t &x) const {
+    if (contains(x)) {
+      return get_all_members_from_an_equiv_class(x);
+    } else {
+      return boost::none;
+    }
+  }
+
+  /// @brief get the equivalent class by giving a domain value
+  /// @param dom a domain value such as #var1
+  /// @return if the value exists, return corresponding element class
+  boost::optional<element_set_t> get_variables(const domain_t &dom) const {
+    for (auto &kv : m_classes) {
+      const element_t &key = kv.first;
+      const equivalence_class_t &ec = kv.second;
+      auto dom_ptr = ec.get_absval();
+      if (dom_ptr && dom == *dom_ptr) {
+        return get_all_members_from_an_equiv_class(key);
+      }
+    }
+    return boost::none;
+  }
+
+  /// @brief get the domain value stored in current equivalent class
+  /// @param v an element in some set
+  /// @return null if !contains(x), otherwise, returns a shared pointer to value
+  std::shared_ptr<const domain_t> get(const element_t &x) const {
+    if (is_bottom()) {
+      CRAB_ERROR("called ", domain_name(), "::", __func__, " on bottom");
+    }
+    if (is_top() || !contains(x)) {
+      return nullptr;
+    }
+
+    element_t rep_x = find(x);
+    const equivalence_class_t &ec_x = m_classes.at(rep_x);
+    return ec_x.get_absval();
+  }
+
+  /// @brief Add y into the equivalence class of x
+  /// @param x an element may exists
+  /// @param y an element may exists
+  /// @note  if x does not exist, nothing changed.
+  ///        if y \in some cls, forget it from cls, add y into x's class
+  void add(const element_t &x, const element_t &y) {
+    if (is_bottom() || is_top() || x == y) {
+      return;
+    }
+    if (!contains(x)) {
+      set(x, symbolic_variable_equiality_domain_impl::make_fresh_var_symbol<
+                 symbolic_variable_equiality_domain_impl::symbolic_var>());
+    }
+    if (contains(y)) {
+      *this -= y;
+    }
+    element_t rep_x = find(x);
```

Comment:
This should be done by calling the `union` operation. You are not updating the size of the class